### PR TITLE
Fix superpmi.py/jitrollingbuild.py to handle large REST results

### PR DESCRIPTION
**Pull request by BruceForstall**
The Azure Storage REST API returns a maximum of 5000 results in
a single query. We have more results in the JIT rolling build set,
so we weren't finding queried git_hash values.

Generalize the query to loop using the "marker" continuation functionality.

We currently have about 800 superpmi collections results, but update
the superpmi.py script similarly for future proofing.

I also changed the query used by jitrollingbuild.py to specify a prefix,
to avoid requiring so many results. Probably should do the same thing for
SuperPMI collections. The downside is the REST api doesn't have a facility
for the prefix to be case-insensitive. I think it's ok to be case-sensitive,
but we'd need to verify that.

### CODE REVIEW

**Comment by BruceForstall** [comment]
@kunalspathak @dotnet/jit-contrib PTAL

File: src/coreclr/scripts/superpmi.py [hunk 1]
```diff
@@ -2407,45 +2407,69 @@ def list_superpmi_collections_container_via_rest_api(path_filter=lambda unused:
...
+
+        urls_split = contents.split("<Url>")[1:]
+        for item in urls_split:
+            url = item.split("</Url>")[0].strip()
+            path = remove_prefix(url, url_prefix)
+            if path_filter(path):
+                paths.append(path)
+
+        # Look for a continuation marker.
+        re_match = re.match(r'.*<NextMarker>(.*)</NextMarker>.*', contents)
```
Comment (kunalspathak) [inline src/coreclr/scripts/superpmi.py:2466]:
```suggestion
        re_match = re.match(r'<NextMarker>(.*)</NextMarker>', contents)
```

File: src/coreclr/scripts/superpmi.py [hunk 2]
```diff
@@ -2407,45 +2407,69 @@ def list_superpmi_collections_container_via_rest_api(path_filter=lambda unused:
...
+        if re_match:
+            marker_text = re_match.group(1)
+            marker = "&marker=" + marker_text
+            iter += 1
```
Comment (kunalspathak) [inline src/coreclr/scripts/superpmi.py:2470]: Is `iter` used anywhere? I don't see anything like that.

File: src/coreclr/scripts/jitrollingbuild.py [hunk 3]
```diff
@@ -187,46 +188,72 @@ def list_az_jits(filter_func=lambda unused: True):
...
+        # use regular expressions.
+
+        urls_split = contents.split("<Url>")[1:]
+        for item in urls_split:
+            url = item.split("</Url>")[0].strip()
+            if filter_func(url):
+                urls.append(url)
+
+        # Look for a continuation marker.
+        re_match = re.match(r'.*<NextMarker>(.*)</NextMarker>.*', contents)
```
Comment (kunalspathak) [inline src/coreclr/scripts/jitrollingbuild.py:250]: May be secondary priority to import this function from other script? Same suggestions are applicable here though

**Review by kunalspathak — APPROVED** [review]
LGTM with minor suggestions.

File: src/coreclr/scripts/superpmi.py (hunk 2 above)
Comment (BruceForstall) [reply src/coreclr/scripts/superpmi.py:2470]: No, I used it for debugging, and left it in.

File: src/coreclr/scripts/jitrollingbuild.py (hunk 3 above)
Comment (BruceForstall) [reply src/coreclr/scripts/jitrollingbuild.py:250]: I would be nice to share more functions between the two files... eventually...

**Pull request merged**